ui/raidboss/data/02-arr/dungeon/haukke_manor.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+[{
+  zoneId: ZoneId.HaukkeManor,
+  triggers: [
+    {
+      id: 'Haukke Normal Dark Mist DPS/Heals',
+      netRegex: NetRegexes.startsUsing({ id: '2C1', source: ['Manor Maidservant', 'Manor Claviger', 'Lady Amandine'], capture: false }),
+      condition: function(data) {
+        return data.role != 'tank';
+      },
+      suppressSeconds: 2,
+      response: Responses.getOut('alert'),
+    },
+    {
+      id: 'Haukke Normal Dark Mist Tank',
+      netRegex: NetRegexes.startsUsing({ id: '2C1', source: ['Manor Maidservant', 'Manor Claviger', 'Lady Amandine'] }),
+      condition: function(data) {
```

Comment:
Here and elsewhere, both for situations where you need to stun and for interrupts, you're better off using the `data.canStun()` or `data.canSilence()` construction, rather than just restricting it to tanks.

Also, is Amandine stun-vulnerable? I haven't tested it myself, but I always thought final bosses were immune even in ARR.

Reply:
After logging 5-10 runs so that I could create timelines, during rigorous testing to ensure matching once they had passed test_timeline satisfactorily, I learned that the bosses in there often delayed their abilities. I would assume this was because they have so few as Haukke Normal level, but most of the time, timelines seemed to be off by 1.5-2 seconds at various intervals throughout the fight, and they seemed random from pull to pull, so I just scrapped the timelines altogether until I get to something higher that's a bit more scripted.

Reply:
Also, for your comment about Amandine, I will double-check her stun vulnerability right now; actually, you bring up a good point. You are correct about changing to data.canStun(),. I didn't do it at first because I wasn't sure if all the roles had their stuns and interrupts by this level, but it turns out they all do, so I will go ahead and change that.

In terms of not triggering for visible AoEs, I will remove the generic all instance triggers for Dark Mist for anyone without stuns and the trigger for Soul Drain as I would like to keep this consistent with the rest of the repo. I want to make a separate Dark Mist Trigger that applies to everyone that is restricted to Lady Amandine, though. With all the particle effects in her room and all the spell effects from players, sometimes you can't see her Dark Mist AoE at all if she's sitting in the fog at the edges of the room for whatever reason.  This mostly seems to depend on where she is being tanked, whether it be for a reason or simply because the tank is a new player. I want to keep a Dark Mist trigger for Amandine in, though, simply to alleviate issues of possibly getting hit by a mechanic you should be able to see but can't due to room graphics.

Reply:
Amandine is stun vulnerable by the way.

Reply:
Okay, yes, thanks for removing those others. I thought about it more today and talked it over, and given that we aren't doing that except for specific situations elsewhere in the repository, it doesn't make sense to start here for the non-Amandine ones. @quisquous will probably have to weigh in on Amandine, but definitely for the future avoid it unless there's a specific reason for it. (An example of where I argued for this is on [Sophia EX](https://github.com/quisquous/cactbot/blob/0712f53cdde5ae5a8e49fe22cf82c2c8a9047d45/ui/raidboss/data/03-hw/trial/sophia-ex.js#L142), as a reference point.)

Reply:
I made a mistake in the casing on the conditions there. It should be `CanStun` or `CanSilence`. See the [raidboss config](https://github.com/quisquous/cactbot/blob/672f8e949bd03643c234ad453875eac562b75a33/ui/raidboss/raidboss_config.js#L375) for the full selection. (Also, it's `CanSilence`, not `CanInterrupt`, because these utility functions were added in Stormblood when that was the terminology.)

I think once you fix those you should see your triggers working. Or, at the very least, when I swapped these out, It Worked On My Machine.